### ISSUE TITLE
Utilisation de Safe Exam 

### ISSUE THREAD

**Issue by sfraysse** [sfraysse/moodle-trainingpath#7]
Etudier les possibilités d'exploitation de Safe Exam avec les activités SCORM Lite, AssessmentPath et TrainingPath.

**Comment by sfraysse** [comment]
Après un rapide tour d'horizon, il faut distinguer plusieurs aspects :

**1. Interdire l'accès à certaines pages lorsque le navigateur SafeExam n'est pas utilisé**. C'est à priori assez simple. Il faudra insérer une vérification sur les-dites pages. Ca peut être très ciblé (ex. seules les pages des tests d'un Training Path, pour lesquels une nouvelle option "secure" serait ajoutée). https://moodle.org/mod/forum/discuss.php?d=232566

**2. Faire en sorte que la navigation générale Moodle soit masquée**. Il faut pour cela utiliser un thème compatible, c'est à dire qui fournit un modèle de page sécurisée "secure.php". A priori, c'est le cas des thèmes Boost et Clean. Il faut ensuite modifier nos plugins pour faire appel à ce modèle de page lorsque nécessaire.

**3. Faire en sorte que la navigation interne à l'activité soit masquée**. Il faut revoir la composition des pages à sécuriser pour offrir 2 options de présentation : présentation actuelle, et une version sans navigation interne. Par exemple sur TrainingPath : masquer les onglets, le fil d'ariane, etc. Mais s'il ne s'agit que de la page du Player SCORM, ces éléments sont déjà masqués.

**4. Sécuriser les popup**. C'est un peu plus flou à ce stade mais il semble qu'il faille aussi traiter le cas de l'affichage fenêtré qui semble poser des problèmes spécifiques.

**Comment by sfraysse** [comment]
**Concernant les popup**, en effet, elles posent un problème avec Safe Exam. Elles sont théoriquement supportées, mais se retrouvent derrière la fenêtre principale qui est en plein écran (tout du moins sur mon poste, avec la dernière version de SEM).

Ceci dit, le mode popup n'est pas vraiment utile car sans popup, le contenu est déjà en quasi plein écran du fait que toute la nav soit masquée. Donc la préco **avec Safe Exam est de ne pas afficher le contenu dans une popup**.

**Comment by sfraysse** [comment]
Mis en oeuvre sur SCORMLite : https://github.com/sfraysse/moodle-scormlite/issues/5

**Comment by stephanebarcet** [comment]
Vu dans scormlite.
Du coup, est-il possible d'intégrer ce paramètre dans l'Assessment path et le Training path ?

**Comment by sfraysse** [comment]
Oui, on va pouvoir l'intégrer. 
Je n'ai pas encore évalué le temps nécessaire mais je n'ai pas rencontré de grande difficulté sur ScormLite, donc ça devrait être assez rapide. 
Je te propose de laisser ce point ouvert pour le moment.

**Comment by stephanebarcet** [comment]
OK

**Comment by stephanebarcet** [comment]
Effectivement, on rode ca sur scormlite, pour le pousser plus tard sur les autres modules.

**Comment by sfraysse** [comment]
Le fonctionnement de Safe Exam a changé pour la version 3.9 de Moodle.

**Comment by sfraysse** [comment]
Le fonctionnement de SEB a "sauté" avec Moodle 3.9 et il y a un risque qu'elle "saute" à nouveau avec Moodle 4.0 qui touche en profondeur à l'UX, donc potentiellement à l'intégration avec Safe Exam Browser. Mon conseil est d'attendre la sortie de Moodle 4.0 avant de "réparer" ou étendre cette fonction.
